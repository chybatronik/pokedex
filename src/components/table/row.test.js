import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import Row from './row'
import expect from 'expect'

Enzyme.configure({ adapter: new Adapter() })

it('Row component with isFetch: true', () => {
  const item = {name: 'test', index: 100, isFetch: true}
  const output = shallow(
    <Row
      {...item}
    />
  )
  expect(shallowToJson(output)).toMatchSnapshot()
})

it('Row component with isFetch: false', () => {
  const item = {
    name: 'test',
    index: 100,
    isFetch: false,
    sprites: {},
    types: [],
    weight: 1,
    height: 2
  }
  const output = shallow(
    <Row
      {...item}
    />
  )
  expect(shallowToJson(output)).toMatchSnapshot()
})
