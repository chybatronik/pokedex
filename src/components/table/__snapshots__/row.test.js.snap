// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component with isFetch: false 1`] = `
<tr
  key="test"
>
  <th
    scope="row"
    width="10%"
  >
     
    100
     
  </th>
  <td
    style={
      Object {
        "padding": 0,
      }
    }
    width="10%"
  >
    <img
      alt="test"
      async={true}
      height={52}
    />
  </td>
  <td>
    test
  </td>
  <td>
    <TagsComponent
      types={Array []}
    />
  </td>
  <td
    width="10%"
  >
    1
  </td>
  <td
    width="10%"
  >
    2
  </td>
</tr>
`;

exports[`Row component with isFetch: true 1`] = `
<tr
  key="test"
>
  <th
    scope="row"
    width="10%"
  >
     
    100
     
  </th>
  <td
    width="10%"
  >
    <Spinner
      color="blue"
      fadeIn="full"
      name="circle"
      noFadeIn={false}
      overrideSpinnerClassName=""
    />
  </td>
  <td>
    test
  </td>
  <td>
    <TagsComponent />
  </td>
  <td
    width="10%"
  />
  <td
    width="10%"
  />
</tr>
`;
